### PR TITLE
unittest: Fail with big-edian architecture 

### PR DESCRIPTION
**Pull request by LocutusOfBorg**
Swap some bytes makes it work on ppc64 and s390x

See: https://github.com/COVESA/dlt-daemon/issues/523

### CODE REVIEW

**Comment by minminlittleshrimp** [comment]
Okay let me check the change with s390x/debian on docker
Regards

**Comment by minminlittleshrimp** [comment]
Hello @LocutusOfBorg ,
thanks for your interest in DLT topic.
I could reproduce successfully in my local qemu.
Honestly, I think this FAIL of the test case is understandable, and as you pointed out, yes, it is BYTE_ORDER issue when the author of the test case was not aware of the Big edian case.
I will upload my fix together with an CI/CD for Big Edian check in the future.
Once again thank you so much.
Minh

File: tests/gtest_dlt_daemon_gateway.cpp [hunk 1]
```diff
@@ -458,6 +458,10 @@ TEST(t_dlt_gateway_parse_get_log_info, normal)
     uint8_t status = 7;
     uint16_t count_app_ids = 1;
     uint16_t count_context_ids = 1;
+#if (BYTE_ORDER == BIG_ENDIAN)
+    count_app_ids = DLT_HTOLE_16(count_app_ids);
+    count_context_ids = DLT_HTOLE_16(count_context_ids);
+#endif
```
Comment (michael-methner) [inline tests/gtest_dlt_daemon_gateway.cpp:464]: Hello @LocutusOfBorg ,
thanks for the patch but these IDs should not be in little endian when the MSBF bit is set.

I think we need rather a fix in the function dlt_getloginfo_conv_ascii_to_uint16_t() to reflect the endianess of the payload (as indicated by the MSBF bit)

**Comment by minminlittleshrimp** [comment]
Close and move discussion to https://github.com/COVESA/dlt-daemon/pull/529

**Pull request closed without merging**